### PR TITLE
Fix & clean up examples

### LINKED ISSUE #73: Fix Example 4
Example 4 is currently not working 100%. I think this might have more to do with the setup instructions than the example itself. Ideally someone with knowledge of Apollo can get it up and running with good docs.

### PR DESCRIPTION
**Pull request by zenflow**
This fixes everything I found broken in the examples and cleans things up a bit too.

Closes #73

### CODE REVIEW

File: examples/1-getting-started/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "start:server": "node ./src/server/index.js",
     "start:client": "sleep 1 && yarn scaffold && parcel serve -p 3000 public/index.html",
-    "scaffold": "../../generator/mst-gql-scaffold.js --roots 'Todo' --excludes 'CacheControlScope,Query,Subscription' --outDir src/app/models/ --format ts http://localhost:3001/graphql",
+    "scaffold": "node ../../generator/mst-gql-scaffold.js --roots Todo --outDir src/app/models/ --format ts http://localhost:3001/graphql",
```
Comment (mattiasewers) [inline examples/1-getting-started/package.json]: This file `mst-gql-scaffold` contains `#!/usr/bin/env node` so there is no need to execute it with node

File: examples/3-twitter-clone/package.json [hunk 2]
```diff
@@ -7,8 +7,8 @@
     "start:server": "yarn scaffold:server && ts-node-dev --transpileOnly --project ./src/server/tsconfig.json ./src/server/index.ts",
     "start:client": "yarn scaffold:client && parcel serve -p 3000 public/index.html",
     "scaffold": "yarn scaffold:server && yarn scaffold:client",
-    "scaffold:client": "../../generator/mst-gql-scaffold.js --excludes 'Mutation,CacheControlScope,Query,Subscription' --roots 'Message,User' --outDir src/app/models/ --format ts src/server/schema.graphql",
-    "scaffold:server": "../../generator/mst-gql-scaffold.js --excludes 'Mutation,CacheControlScope,Query,Subscription' --roots 'Message,User' --outDir src/server/models/ --format ts --modelsOnly src/server/schema.graphql",
+    "scaffold:client": "node ../../generator/mst-gql-scaffold.js --roots Message,User --outDir src/app/models/ --format ts src/server/schema.graphql",
+    "scaffold:server": "node ../../generator/mst-gql-scaffold.js --roots Message,User --outDir src/server/models/ --format ts --modelsOnly src/server/schema.graphql",
```
Comment (mattiasewers) [inline examples/3-twitter-clone/package.json]: Same comment as above

File: examples/4-apollo-tutorial/client/package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     "build": "react-scripts build",
     "test": "react-scripts test",
     "eject": "react-scripts eject",
-    "scaffold": "../../../generator/mst-gql-scaffold.js --excludes 'CacheControlScope,Mutation,PatchSize,Query,TripUpdateResponse' --format js demo_schema.json",
+    "scaffold": "node ../../../generator/mst-gql-scaffold.js --excludes PatchSize,TripUpdateResponse --format js demo_schema.json",
```
Comment (mattiasewers) [inline examples/4-apollo-tutorial/client/package.json]: No need for `node`

File: examples/1-getting-started/package.json (hunk 1 above)
Comment (zenflow) [reply examples/1-getting-started/package.json]: I need the `node ` bit or the script doesn't pick up any of the command-line arguments and tries to run with all the defaults.. If you don't see the same thing happening, the difference might be that I'm running Windows 10? Are you running linux/macos? And does the package script still run as expected *with* the `node ` bit?

Comment (mattiasewers) [reply examples/1-getting-started/package.json]: I'm on both mac and linux and it works fine there so this seems to be an issue with windows then. Which version of node are you on?

Comment (zenflow) [reply examples/1-getting-started/package.json]: I'm on node v10.16.0 & yarn v1.16.0. I'll see if upgrading fixes it for me.

Comment (zenflow) [reply examples/1-getting-started/package.json]: I'm on node v10.1.6.0 and yarn v1.16.0. I'll try upgrading to see if that fixes it for me.

Comment (zenflow) [reply examples/1-getting-started/package.json]: @mattiasewers After upgrading to the latest node LTS (v10.16.3) and latest yarn (v1.17.3) I'm still experiencing the same problem. Does my change break things for you? Or can we keep it for improved portability?

Comment (chrisdrackett) [reply examples/1-getting-started/package.json]: If this is needed to run things on windows and does not break other platforms I feel we should leave it in

File: examples/2-scaffolding/src/models/PokemonAttackModel.ts [hunk 4]
```diff
@@ -2,24 +2,20 @@ import { Instance } from "mobx-state-tree"
 import { PokemonAttackModelBase } from "./PokemonAttackModel.base"
 
 /* The TypeScript type of an instance of PokemonAttackModel */
-export interface PokemonAttackModelType
-  extends Instance<typeof PokemonAttackModel.Type> {}
+export interface PokemonAttackModelType extends Instance<typeof PokemonAttackModel.Type> {}
```
Comment (chrisdrackett) [inline examples/2-scaffolding/src/models/PokemonAttackModel.ts]: @zenflow some of these formatting changes are reverted by prettier for me, do you know if prettier is running pre-commit for you as it should?

Comment (zenflow) [reply examples/2-scaffolding/src/models/PokemonAttackModel.ts]: @chrisdrackett No, it doesn't seem to be running. 

These formatting changes are the result of running `yarn test` which involves `cd`ing into `examples/2-scaffolding` and running the `scaffold` script which uses "force" option, so it looks like prettier is fighting with our generated code here.

But that should be ok if the commit hook works; I'll sort that out.

I'll make a commit now to format everything with prettier.

Comment (chrisdrackett) [reply examples/2-scaffolding/src/models/PokemonAttackModel.ts]: cool! I wonder if something around husky isn't working on windows?

**Comment by chrisdrackett** [comment]
other than regressions around prettier formatting this looks ready to merge for me!

File: examples/2-scaffolding/src/models/PokemonAttackModel.ts (hunk 4 above)
Comment (zenflow) [reply examples/2-scaffolding/src/models/PokemonAttackModel.ts]: lol seems safe to assume that being on windows is the problem 💩

**Comment by zenflow** [comment]
> other than regressions around prettier formatting this looks ready to merge for me!

@chrisdrackett With commit eebde45 I normalized the formatting with prettier (for all files) using the settings from #109 

This should be good to merge

**Pull request merged**